src/sentry/api/endpoints/event_grouping_info.py

```diff
@@ -8,17 +9,20 @@
 from sentry.api.base import region_silo_endpoint
 from sentry.api.bases.project import ProjectEndpoint
 from sentry.api.exceptions import ResourceDoesNotExist
+from sentry.eventstore.models import Event
 from sentry.grouping.api import GroupingConfigNotFound
-from sentry.grouping.variants import PerformanceProblemVariant
+from sentry.grouping.variants import BaseVariant, PerformanceProblemVariant
 from sentry.models.project import Project
 from sentry.utils import json, metrics
 from sentry.utils.performance_issues.performance_detection import EventPerformanceProblem
 
 logger = logging.getLogger(__name__)
 
 
-def get_grouping_info(config_name: str | None, project: Project, event_id: str):
-    event = eventstore.backend.get_event_by_id(project.id, event_id)
+def get_grouping_info(
+    config_name: str | None, project: Project, event_id: str, event: Event | None = None
+):
+    event = event if event else eventstore.backend.get_event_by_id(project.id, event_id)
```

Comment:
To make rebasing easier, I pulled this out into a separate PR: https://github.com/getsentry/sentry/pull/69932.